packages/core/src/scp-cf/destination-accessor.ts

```diff
@@ -153,348 +168,420 @@ export async function getDestinationFromDestinationService(
+      logger.info('Could not retrieve destination from destination service.');
+      return null;
     }
-  }
 
-  const { userJwt, ...optionsWithoutJwt } = options;
-  const providerToken = await serviceToken('destination', optionsWithoutJwt);
-  const decodedProviderJwt = await decodeJwt(providerToken);
+    let { destination } = destinationResult;
+    if (destination.authentication === 'OAuth2SAMLBearerAssertion') {
+      destination = await da.addOAuthSamlAuth(destination);
+    }
+    if (destination.authentication === 'ClientCertificateAuthentication') {
+      destination = await da.addClientCertAuth();
+    }
 
-  if (options.useCache) {
-    providerDestinationCache = destinationCache.retrieveDestinationFromCache(
-      decodedProviderJwt,
-      name,
-      isolation
+    const withProxySetting = await da.addProxyConfiguration(destination);
+    await da.updateDestinationCache(
+      withProxySetting,
+      destinationResult.account
     );
-    if (
-      providerDestinationCache &&
-      (selectionStrategy === alwaysProvider ||
-        (decodedUserJwt &&
-          isIdenticalTenant(decodedUserJwt!, decodedProviderJwt)))
-    ) {
+    return withProxySetting;
+  }
+
+  private static searchServiceBindingForDestination(
+    name: string
+  ): Destination | undefined {
+    logger.info('Attempting to retrieve destination from service binding.');
+    try {
+      const destination = destinationForServiceBinding(name);
+      logger.info('Successfully retrieved destination from service binding.');
+      return destination;
+    } catch (error) {
+      logger.info(error.message);
+      logger.info('Could not retrieve destination from service binding.');
       logger.info(
-        'Successfully retrieved destination from destination service cache for provider destinations.'
+        'If you are not using SAP Extension Factory, this information probably does not concern you.'
+      );
+      return undefined;
+    }
+  }
+
+  private static searchEnvVariablesForDestination(
+    name: string
+  ): Destination | undefined {
+    logger.info(
+      'Attempting to retrieve destination from environment variable.'
+    );
+
+    if (getDestinationsEnvVariable()) {
+      logger.warn(
+        "Environment variable 'destinations' is set. Destinations will be read from this variable. " +
+          'This is discouraged for a productive application! ' +
+          'Unset the variable to read destinations from the destination service on SAP Cloud Platform.'
       );
-      return providerDestinationCache;
+
+      try {
+        const destination = getDestinationFromEnvByName(name);
+        if (destination) {
+          logger.info(
+            'Successfully retrieved destination from environment variable.'
+          );
+          return destination;
+        }
+      } catch (error) {
+        logger.info(error.message);
+      }
     }
+
+    logger.info('Could not retrieve destination from environment variable.');
```

Comment:
This is quite misleading. When we get here we didn't even try. Consider something like this:
```suggestion
    logger.info('No environment variable set.');
```